pulp_container/app/models.py

```diff
@@ -633,3 +634,4 @@ class Upload(CoreUpload):
     """
 
     repository = models.ForeignKey(Repository, related_name="uploads", on_delete=models.CASCADE)
+    artifact = models.ForeignKey(Artifact, related_name="uploads", null=True, on_delete=models.PROTECT)
```

Comment:
@mdellweg I am hesitant to add `protect` here.
In case orphan clean up will run, task will fail with `ProtectedError: ("Cannot delete some instances of model 'Artifact' because they are referenced through protected foreign keys: 'Upload.artifact'.", {<Upload: pk=9a0acf29-0371-4f72-af3f-8f5c9667173e>})
`
In core were are checking the timestamp and the content membership so it would be identified as `orphaned` given that the timestamp will be small.

I think I am ok not to put protect here just because the time between artifact creation in patch call and put call this line https://github.com/pulp/pulp_container/pull/810/files#diff-1b944e15118f82233aa03a435e9e05c188ad0d371b71be43260506b4a5eeddffR709 would be few seconds. Obviously this is a window but it should be ok unless orphan_protection time is set to 0 or few seconds. WDYT?

Reply:
We can still "touch" the artifact, right? Then this should be ok.